tests/test_model_saving.py

```diff
@@ -708,6 +708,22 @@ def test_saving_constant_initializer_with_numpy():
     os.remove(fname)
 
 
+def test_saving_group_naming_h5py():
+    """Test saving model with layer which name is prefix to a previous layer
+    name
+    """
+
+    input_layer = Input((None, None, 3), name='test_input')
+    x = Conv2D(1, 1, name='conv1/conv')(input_layer)
+    x = Activation('relu', name='conv1')(x)
+
+    model = Model(inputs=input_layer, outputs=x)
+    _, fname = tempfile.mkstemp('.h5')
+    model.save_weights(fname)
+    model.load_weights(fname)
+    os.remove(fname)
```

Comment:
In your case, you can use the `tmp_path` fixture, which will ensure temporary files are cleaned up, even in case of failure. Please take a look at this: http://doc.pytest.org/en/latest/tmpdir.html#the-tmp-path-fixture  

When using the fixture, you don't have to import it, nor to remove the directory afterwards, pytest takes care of it.

Reply:
Thanks for the info, I'm changing it